frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourceLinkList.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { ResourceLink } from '@console/internal/components/utils';
+import { PersistentVolumeClaimModel, ConfigMapModel, SecretModel } from '@console/internal/models';
+import {
+  PipelineRunWorkspace,
+  VolumeTypeConfigMaps,
+  VolumeTypePVC,
+  VolumeTypeSecret,
+  VolumeTypeClaim,
+} from '../../../types';
+import VolumeClaimTemplateLink from './VolumeClaimTemplateLink';
+
+export interface WorkspaceResourceLinkListProps {
+  workspaces: PipelineRunWorkspace[];
+  namespace: string;
+}
+
+const WorkspaceResourceLinkList: React.FC<WorkspaceResourceLinkListProps> = ({
+  workspaces,
+  namespace,
+}) => {
+  const { t } = useTranslation();
+  return workspaces?.length > 0 ? (
```

Comment:
Nit: Ideally for cleaner code, you should just return early instead of making a return into a large ternary statement.